app/jobs/scheduled/aggregate_web_hooks_events.rb

```diff
@@ -4,10 +4,16 @@ module Jobs
   class AggregateWebHooksEvents < ::Jobs::Scheduled
     every 1.day
 
-    def execute(args)
-      WebHook.all.each do |web_hook|
-        WebHookEventsDailyAggregate.create!(web_hook_id: web_hook.id, date: Time.zone.now.to_date)
-      end
+    def execute(args = {})
+      date = args[:date].present? ? Date.parse(args[:date]) : Time.zone.now.to_date
```

Comment:
You should add a spec for this and the `.where.not(web_hook_events_daily_aggregates: { date: date })` behaviour for sure, checking that:

1. If the date is present, the correct aggregate data is created
2. If the aggregate data already exists for that date, a duplicate record is not created